### PR TITLE
feat: add JSON transport

### PR DESCRIPTION
**Pull request by marcus-j-davies**
So,

This is my first real stab at Typescript.

there might be 1 or 2 things, I didn't get right (or check)

be gentle 😄

### CODE REVIEW

File: packages/json/src/index.ts
Comment (AlCalzone) [inline packages/json/src/index.ts:37]: I think you need to overwrite the `close` method of `Transport`. At least the built-in file transport does this:
https://github.com/winstonjs/winston/blob/2625f60c5c85b8c4926c65e98a591f8b42e0db9a/lib/winston/transports/file.js#L454

File: packages/json/src/index.ts
Comment (AlCalzone) [inline packages/json/src/index.ts:14]: AFAIK this MUST be `"silly"` or changes to zwave-js's loglevel won't propagate properly.

File: packages/json/src/index.ts
Comment (AlCalzone) [inline packages/json/src/index.ts:11]: No need for this to be a class member, just make it a const in the module scope

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Comment by AlCalzone** [comment]
@raman325 What the point of having `formattedMessage` in addition to `message`?
formatted message is date + label + direction + primary tags + message + padding + secondary tags

If you want the formatted message, we could either add a property to the constructor to add the default format and overwrite the `message` with the symbol property one. Or we could leave it up to the transport consumer and overwrite `message` when the symbol property exists.

**Comment by AlCalzone** [comment]
Also, if you use objectMode streams and skip the stringify, symbol properties will be included in the object:
grafik
(this is with default formatting:)
```ts
j.format = createDefaultTransportFormat(false, false);
```

**Review by AlCalzone — APPROVED** [review]
I'll wait for feedback on the formatted message, but I'm okay with this now

**Comment by marcus-j-davies** [comment]
currently using my own implementation, but will switch over to this one, once.
is this actually bundled with the driver?

**Comment by AlCalzone** [comment]
No, each transport is a separate package you have to install yourself.

**Comment by raman325** [comment]
> @raman325 What the point of having `formattedMessage` in addition to `message`?
> formatted message is date + label + direction + primary tags + message + padding + secondary tags
> 
> If you want the formatted message, we could either add a property to the constructor to add the default format and overwrite the `message` with the symbol property one. Or we could leave it up to the transport consumer and overwrite `message` when the symbol property exists.

That's a fair point. To me `formattedMessage` > `message` because it allows the consumer to show the user the actual log message vs trying to recreate it from all the input parameters

**Pull request merged**
